File: Sock_Merchant.py

def sockMerchant(n,ar):
    c = 0
    for i in set(ar):
        c += ar.count(i) // 2

    return c

n = int(input())
ar = list(map(int, input().rstrip().split()))
result = sockMerchant(n, ar)
print(result)

File: Sock_Merchant.cpp
#include <bits/stdc++.h>
#include <set>
using namespace std;

int sockMerchant(int n, vector<int> ar) {
    set<int> s1{begin(ar), end(ar)};
    int cnt = 0;
    for(auto const v:s1)
    {
        cnt += int(count(ar.begin(), ar.end(), v)/2);
    }
    return cnt;
}

int main()
{
    int n;
    cin >> n;
    int temp;
    vector<int> ar;

    for (int i = 0; i < n; i++) {
        cin >> temp;
        ar.push_back(temp);
    }

    int result = sockMerchant(n, ar);
    cout << result << endl;
    return 0;
}
